root/static/scripts/common/utility/formatDate.js

```diff
@@ -23,22 +22,22 @@ function formatDate(date: PartialDateT | null): string {
 
   if (nonEmpty(y)) {
     if (y < 0) {
-      result += '-' + padStart(String(-y), 3, '0');
+      result += '-' + String(-y).padStart(3, '0');
```

Comment:
Added a `fixedWidthInteger` utility. It handles negative numbers too (here for example) by making sure the `-` stays in front of the zeroes. This actually changed the `formatDate` output slightly for negative years; it now outputs year -1 like `-0001` instead of `-001`. But I think that's fine?